iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -139,39 +141,44 @@ private async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri,
                         // Configure proxy server
                         websocket.Options.Proxy = _amqpTransportSettings.Proxy;
                         if (Logging.IsEnabled)
-                            Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Set ClientWebSocket.Options.Proxy to {_amqpTransportSettings.Proxy}");
+                            Logging.Info(this, $"{nameof(CreateClientWebSocket)} Set ClientWebSocket.Options.Proxy to {_amqpTransportSettings.Proxy}");
                     }
                     catch (PlatformNotSupportedException)
                     {
                         // Some .NET runtimes don't support this property.
                         if (Logging.IsEnabled)
-                            Logging.Error(this, $"{nameof(CreateClientWebSocketAsync)} PlatformNotSupportedException thrown as this framework doesn't support proxy.");
+                            Logging.Error(this, $"{nameof(CreateClientWebSocket)} PlatformNotSupportedException thrown as this framework doesn't support proxy.");
+                        throw new InvalidOperationException("The current .NET runtime does not support setting the proxy.");
                     }
                 }
 
                 if (_amqpTransportSettings.WebSocketKeepAlive.HasValue)
                 {
                     websocket.Options.KeepAliveInterval = _amqpTransportSettings.WebSocketKeepAlive.Value;
                     if (Logging.IsEnabled)
-                        Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Set websocket keep-alive to {_amqpTransportSettings.WebSocketKeepAlive}");
+                        Logging.Info(this, $"{nameof(CreateClientWebSocket)} Set websocket keep-alive to {_amqpTransportSettings.WebSocketKeepAlive}");
                 }
 
                 if (_connectionCredentials.Certificate != null)
                 {
                     websocket.Options.ClientCertificates.Add(_connectionCredentials.Certificate);
                 }
 
-                await websocket.ConnectAsync(websocketUri, cancellationToken).ConfigureAwait(false);
-
                 return websocket;
             }
             finally
             {
                 if (Logging.IsEnabled)
-                    Logging.Exit(this, nameof(CreateClientWebSocketAsync));
+                    Logging.Exit(this, nameof(CreateClientWebSocket));
             }
         }
 
+        /// <summary>
+        /// Remote certificate validation callback.
+        /// </summary>
+        /// <remarks>
+        /// Remote certificate validation is not supported for web socket implementation.
+        /// </remarks>
```

Comment:
I'd add a note explaining how to get this feature through providing a client web socket instance. See my PR #2829 for how I wrote this up